### DISCUSSION TITLE
[v3] Livewire.js error preventing component from updating

### DISCUSSION THREAD

**Discussion by DougSisk** [livewire/livewire#5995] [7. Bugs (archived)] [answered]
Unsure what's going on, but I'm getting an error with Livewire.js that's preventing one of my components from updating.

```
livewire.js?id=578b80d0:7559
Uncaught TypeError: Cannot read properties of undefined (reading 'nodeType')
    at isIf (livewire.js?id=578b80d0:7559:35)
    at patchChildren (livewire.js?id=578b80d0:7567:17)
    at patch (livewire.js?id=578b80d0:7488:7)
    at patchChildren (livewire.js?id=578b80d0:7641:9)
    at patch (livewire.js?id=578b80d0:7488:7)
    at patchChildren (livewire.js?id=578b80d0:7641:9)
    at patch (livewire.js?id=578b80d0:7488:7)
    at patchChildren (livewire.js?id=578b80d0:7641:9)
    at patch (livewire.js?id=578b80d0:7488:7)
    at patchChildren (livewire.js?id=578b80d0:7641:9)
```

**Comment by DougSisk** [comment]
Did some digging and it looks like `isIf` is in [the Alpine morph plugin](https://github.com/alpinejs/alpine/blob/cd9d8d35d9e67c7c8722135c7e6dc2a8d6868615/packages/morph/src/morph.js#L171)

**Comment by danie-ramdhani** [comment]
it happens when using the blade component like this:

```blade
@if ($someCondition)
    <div> {{-- don't do this --}}
@endif

...another elements

@if ($someCondition)
    </div> {{-- and this --}}
@endif
```

this component won't work as intended.

**Reply by DougSisk** [reply]
Don't have these conditions in my templates.

**Comment by lrljoe** [comment]
Also happens if you are missing keys in a loop, or have empty elements. Took me an age to find my empty element!

**Reply by DougSisk** [reply]
I've added wire:key to all my loops with unique names and I'm still getting the error.

**Reply by lrljoe** [reply]
Try on the direct parent element, setting the wire key to a prefix , and the child primary keys joined.

That will force a re-render of the element when the list of child items changes.

Check for any empty elements elsewhere as there's a bug where it will cause an error.

**Reply by DougSisk** [reply]
Thanks for your help; ended up being a bad Blade condition. Still going to try my best to remember to use `wire:key`

**Reply by lrljoe** [reply]
No problem, might be worth checking out bladestan, but I'm not sure if its Livewire friendly yet!

**Comment by DougSisk** [comment]
In one of my affected components I was able to narrow it down to a `<datalist>` using unclosed `<option>`. Really annoying v2 didn't have an issue but v3 now does.

**Comment by DougSisk** [comment, marked as answer]
Final affected component had a bad closing condition: `@endauth` instead of `@endif`. All is well now!
